clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -90,7 +90,7 @@ public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
       mUfsPath = fileInfo.getUfsPath();
       String fileName = PathUtils.temporaryFileName(fileId, mNonce, mUfsPath);
       UnderFileSystem ufs = UnderFileSystem.get(fileName, ClientContext.getConf());
-      String parentPath = (new TachyonURI(mUfsPath)).getParent().getPath();
+      String parentPath = (new TachyonURI(mUfsPath)).getParent().toString();
       if (!ufs.exists(parentPath) && !ufs.mkdirs(parentPath, true)) {
```

Comment:
The cause for failure of `tachyon runTests` when ufs is S3 is that the original code didn't preserve ufs path's scheme(like "s3n://").

Say, `parentPath` is "/a/b" in S3, the key for this directory object will be "a/b"(created by S3UnderFileSystem), but `ufs.exists(parentPath)` will try to find an object with key "/a/b", so `exists` fails. But with scheme, parentPath will be "s3n://a/b", then [this method](https://github.com/amplab/tachyon/blob/master/underfs/s3/src/main/java/tachyon/underfs/s3/S3UnderFileSystem.java#L529) will strip "s3n://", and query S3 via key "a/b", then `exists` succeeds.